spec/unit/network/http/connection_spec.rb

```diff
@@ -138,164 +119,131 @@ def expects_limit_exceeded(conn)
       }.to raise_error(Puppet::Network::HTTP::RedirectionLimitExceededException)
     end
 
-    it "should not redirect when the limit is 0" do
-      conn = create_connection(site, :verify => verify, :redirect_limit => 0)
-
-      pool = expects_redirection(conn)
-      expect(pool).not_to receive(:with_connection).with(other_site, anything)
+    it "should not follow any redirects when the limit is 0" do
+      stub_request(:get, "http://me.example.com:8140/").to_return(redirect_to("http://me.example.com:8140/foo"))
 
+      conn = create_connection(:redirect_limit => 0)
       expects_limit_exceeded(conn)
     end
 
-    it "should redirect only once" do
-      conn = create_connection(site, :verify => verify, :redirect_limit => 1)
-
-      pool = expects_redirection(conn)
-      expect(pool).to receive(:with_connection).with(other_site, anything).once
+    it "should follow the redirect once" do
+      stub_request(:get, "http://me.example.com:8140/").to_return(redirect_to("http://me.example.com:8140/foo"))
+      stub_request(:get, "http://me.example.com:8140/foo").to_return(redirect_to("http://me.example.com:8140/bar"))
 
+      conn = create_connection(:redirect_limit => 1)
       expects_limit_exceeded(conn)
     end
 
     it "should raise an exception when the redirect limit is exceeded" do
-      conn = create_connection(site, :verify => verify, :redirect_limit => 3)
-
-      pool = expects_redirection(conn)
-      expect(pool).to receive(:with_connection).with(other_site, anything).exactly(3).times
+      stub_request(:get, "http://me.example.com:8140/").to_return(redirect_to("http://me.example.com:8140/foo"))
+      stub_request(:get, "http://me.example.com:8140/foo").to_return(redirect_to("http://me.example.com:8140/bar"))
+      stub_request(:get, "http://me.example.com:8140/bar").to_return(redirect_to("http://me.example.com:8140/baz"))
+      stub_request(:get, "http://me.example.com:8140/baz").to_return(redirect_to("http://me.example.com:8140/qux"))
 
+      conn = create_connection(:redirect_limit => 3)
       expects_limit_exceeded(conn)
     end
```

Comment:
I would like to see a test where there is a redirection limit, but the limit is not exceeded, and the connection is OK.

Reply:
That's tested in https://github.com/puppetlabs/puppet/pull/7767/files/1c63f9d3231128ceac96fbee00a474942b67ca67#diff-f12903d4e996470e4f3649e6a5d6eb5eR109, as the default redirect limit is 10